### PR TITLE
Javadoc for some client, datafixer and other miscellaneous areas.

### PR DESCRIPTION
**Pull request by i509VCB**

### CODE REVIEW

File: mappings/net/minecraft/client/MinecraftClient.mapping [hunk 1]
```diff
@@ -182,6 +182,9 @@ CLASS net/minecraft/class_310 net/minecraft/client/MinecraftClient
 	METHOD method_1573 forcesUnicodeFont ()Z
 	METHOD method_1574 tick ()V
 	METHOD method_1576 getServer ()Lnet/minecraft/class_1132;
+		COMMENT Gets this client's own integrated server.
+		COMMENT
+		COMMENT The integrated server is only present when a local single player world is open.
```
Comment (liach) [inline mappings/net/minecraft/client/MinecraftClient.mapping]:
```suggestion
		COMMENT <p>The integrated server is only present when a local single player world is open.
```

File: mappings/net/minecraft/client/network/PlayerListEntry.mapping [hunk 2]
```diff
@@ -17,6 +17,9 @@ CLASS net/minecraft/class_640 net/minecraft/client/network/PlayerListEntry
 	METHOD method_2963 setGameMode (Lnet/minecraft/class_1934;)V
 	METHOD method_2966 getProfile ()Lcom/mojang/authlib/GameProfile;
 	METHOD method_2967 hasSkinTexture ()Z
+		COMMENT Checks if the player represented by this entry has a custom skin.
+		COMMENT
+		COMMENT <p>If the player has the default {@code Steve} or {@code Alex} skin, this will return false.
```
Comment (liach) [inline mappings/net/minecraft/client/network/PlayerListEntry.mapping]: "Steve" and "Alex" isn't code.

File: mappings/net/minecraft/command/arguments/TestClassArgumentType.mapping [hunk 3]
```diff
@@ -1,4 +1,6 @@
 CLASS net/minecraft/class_4526 net/minecraft/command/arguments/TestClassArgumentType
+	FIELD field_20580 examples Ljava/util/Collection;
```
Comment (liach) [inline mappings/net/minecraft/command/arguments/TestClassArgumentType.mapping]: if this is `static final` per yarn rule it should be
```suggestion
	FIELD field_20580 EXAMPLES Ljava/util/Collection;
```
even though the field may not be effectively constant.

File: mappings/net/minecraft/datafixer/TypeReferences.mapping [hunk 4]
```diff
@@ -1,26 +1,52 @@
...
 	FIELD field_5723 ENTITY_TREE Lcom/mojang/datafixers/DSL$TypeReference;
+		COMMENT A type reference which refers to an entity tree.
+		COMMENT
+		COMMENT <p>An entity tree contains the passengers of an entity and their passengers.
 	FIELD field_5724 STRUCTURE_FEATURE Lcom/mojang/datafixers/DSL$TypeReference;
 	FIELD field_5725 ADVANCEMENTS Lcom/mojang/datafixers/DSL$TypeReference;
 	FIELD field_5726 CHUNK Lcom/mojang/datafixers/DSL$TypeReference;
+		COMMENT A type reference which refers to a chunk.
 	FIELD field_5727 BLOCK_ENTITY Lcom/mojang/datafixers/DSL$TypeReference;
+		COMMENT A type reference which refers to a block entity.
```
Comment (liach) [inline mappings/net/minecraft/datafixer/TypeReferences.mapping:38]: Is this a block entity or a block entity type?

File: mappings/net/minecraft/datafixer/TypeReferences.mapping [hunk 5]
```diff
@@ -1,26 +1,52 @@
...
 	FIELD field_23067 CHUNK_GENERATOR_SETTINGS Lcom/mojang/datafixers/DSL$TypeReference;
+		COMMENT A type reference which refers to chunk generator settings.
 	FIELD field_5710 LEVEL Lcom/mojang/datafixers/DSL$TypeReference;
 	FIELD field_5711 RECIPE Lcom/mojang/datafixers/DSL$TypeReference;
 	FIELD field_5712 ITEM_STACK Lcom/mojang/datafixers/DSL$TypeReference;
+		COMMENT A type reference which refers to an item stack.
 	FIELD field_5713 ITEM_NAME Lcom/mojang/datafixers/DSL$TypeReference;
+		COMMENT A type reference which refers to an item's name.
+		COMMENT
+		COMMENT <p>The name refers to the identifier this item is registered to.
```
Comment (liach) [inline mappings/net/minecraft/datafixer/TypeReferences.mapping]:
```suggestion
		COMMENT A type reference which refers to an item's identifier.
```

File: mappings/net/minecraft/datafixer/TypeReferences.mapping [hunk 6]
```diff
@@ -1,26 +1,52 @@
...
 	FIELD field_5715 PLAYER Lcom/mojang/datafixers/DSL$TypeReference;
+		COMMENT A type reference which refers to a player.
 	FIELD field_5716 STRUCTURE Lcom/mojang/datafixers/DSL$TypeReference;
 	FIELD field_5717 OPTIONS Lcom/mojang/datafixers/DSL$TypeReference;
+		COMMENT A type reference which refers to client game options.
 	FIELD field_5718 UNTAGGED_SPAWNER Lcom/mojang/datafixers/DSL$TypeReference;
 	FIELD field_5719 ENTITY_NAME Lcom/mojang/datafixers/DSL$TypeReference;
+		COMMENT A type reference which refers to a entity's name.
+		COMMENT
+		COMMENT <p>The name refers to the identifier this entity is registered to.
```
Comment (liach) [inline mappings/net/minecraft/datafixer/TypeReferences.mapping]:
```suggestion
		COMMENT A type reference which refers to a entity's identifier.
```

File: mappings/net/minecraft/datafixer/TypeReferences.mapping [hunk 7]
```diff
@@ -1,26 +1,52 @@
...
+		COMMENT A type reference which refers to a block entity.
 	FIELD field_5728 BIOME Lcom/mojang/datafixers/DSL$TypeReference;
+		COMMENT A type reference which refers to a biome.
 	FIELD field_5729 ENTITY Lcom/mojang/datafixers/DSL$TypeReference;
+		COMMENT A type reference which refers to a type of entity.
 	FIELD field_5730 STATS Lcom/mojang/datafixers/DSL$TypeReference;
 	FIELD field_5731 BLOCK_NAME Lcom/mojang/datafixers/DSL$TypeReference;
+		COMMENT A type reference which refers to a block's name.
+		COMMENT
+		COMMENT <p>The name refers to the identifier this block is registered to.
```
Comment (liach) [inline mappings/net/minecraft/datafixer/TypeReferences.mapping]:
```suggestion
		COMMENT A type reference which refers to a block's identifier.
```

File: mappings/net/minecraft/nbt/NbtHelper.mapping [hunk 8]
```diff
@@ -23,20 +23,31 @@ CLASS net/minecraft/class_2512 net/minecraft/nbt/NbtHelper
 		ARG 1 subject
 		ARG 2 equalValue
 	METHOD method_10688 update (Lcom/mojang/datafixers/DataFixer;Lnet/minecraft/class_4284;Lnet/minecraft/class_2487;I)Lnet/minecraft/class_2487;
+		COMMENT Uses the data fixer to update a tag to the latest data version.
 		ARG 0 fixer
+			COMMENT the data fixer.
```
Comment (liach) [inline mappings/net/minecraft/nbt/NbtHelper.mapping]:
```suggestion
			COMMENT the data fixer
```

File: mappings/net/minecraft/network/listener/ClientPlayPacketListener.mapping [hunk 9]
```diff
@@ -1,4 +1,5 @@
 CLASS net/minecraft/class_2602 net/minecraft/network/listener/ClientPlayPacketListener
+	COMMENT A client side packet listener where packets from the server are processed.
```
Comment (liach) [inline mappings/net/minecraft/network/listener/ClientPlayPacketListener.mapping]: Please mention that this is under the play-stage protocol.

File: mappings/net/minecraft/network/listener/ServerPlayPacketListener.mapping [hunk 10]
```diff
@@ -1,4 +1,5 @@
 CLASS net/minecraft/class_2792 net/minecraft/network/listener/ServerPlayPacketListener
+	COMMENT A server side packet listener where packets from a client are processed.
```
Comment (liach) [inline mappings/net/minecraft/network/listener/ServerPlayPacketListener.mapping]: Also please refer to the play stage protocol.

File: mappings/net/minecraft/server/command/CommandOutput.mapping [hunk 11]
```diff
@@ -1,6 +1,7 @@
 CLASS net/minecraft/class_2165 net/minecraft/server/command/CommandOutput
+	COMMENT Represents a subject which can receive command feedback. 
 	FIELD field_17395 DUMMY Lnet/minecraft/class_2165;
-	METHOD method_9200 sendCommandFeedback ()Z
+	METHOD method_9200 shouldSendCommandFeedback ()Z
```
Comment (liach) [inline mappings/net/minecraft/server/command/CommandOutput.mapping]:
```suggestion
	METHOD method_9200 shouldReceiveFeedback ()Z
```
This is already a "command output", and this output is not sending feedback but receiving feedback

File: mappings/net/minecraft/command/arguments/ArgumentTypes.mapping [hunk 12]
```diff
@@ -7,6 +7,9 @@ CLASS net/minecraft/class_2316 net/minecraft/command/arguments/ArgumentTypes
 	METHOD method_10015 register ()V
 	METHOD method_10016 toJson (Lcom/mojang/brigadier/CommandDispatcher;Lcom/mojang/brigadier/tree/CommandNode;)Lcom/google/gson/JsonObject;
 	METHOD method_10017 register (Ljava/lang/String;Ljava/lang/Class;Lnet/minecraft/class_2314;)V
+		COMMENT Registers an argument type's serializer.
+		ARG 0
```
Comment (liach) [inline mappings/net/minecraft/command/arguments/ArgumentTypes.mapping]:
```suggestion
		ARG 0 id
```
This is apparently causing a build failure.

**Review by liach — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/datafixer/TypeReferences.mapping (hunk 4 above)
Comment (i509VCB) [reply mappings/net/minecraft/datafixer/TypeReferences.mapping:38]: This refers to the block entity itself.

File: mappings/net/minecraft/entity/LivingEntity.mapping [hunk 13]
```diff
@@ -159,7 +160,15 @@ CLASS net/minecraft/class_1309 net/minecraft/entity/LivingEntity
 			COMMENT the main adversary responsible for this entity's death
 	METHOD method_23883 dropXp ()V
 	METHOD method_24518 isHolding (Lnet/minecraft/class_1792;)Z
+		COMMENT Checks if this entity is holding a certain item.
+		COMMENT
+		COMMENT <p>This checks both the entity's main and off hand.
+		ARG 1
```
Comment (liach) [inline mappings/net/minecraft/entity/LivingEntity.mapping]:
```suggestion
		ARG 1 item
```
https://travis-ci.com/github/FabricMC/yarn/builds/155178823#L424
This fails the build

**Review by liach — CHANGES_REQUESTED** [review]

**Review by liach — APPROVED** [review]

File: mappings/net/minecraft/datafixer/TypeReferences.mapping [hunk 14]
```diff
@@ -1,26 +1,46 @@
...
+		COMMENT A type reference which refers to an item's identifier.
 	FIELD field_5714 TEAM Lcom/mojang/datafixers/DSL$TypeReference;
 	FIELD field_5715 PLAYER Lcom/mojang/datafixers/DSL$TypeReference;
+		COMMENT A type reference which refers to a player.
 	FIELD field_5716 STRUCTURE Lcom/mojang/datafixers/DSL$TypeReference;
 	FIELD field_5717 OPTIONS Lcom/mojang/datafixers/DSL$TypeReference;
+		COMMENT A type reference which refers to client game options.
 	FIELD field_5718 UNTAGGED_SPAWNER Lcom/mojang/datafixers/DSL$TypeReference;
 	FIELD field_5719 ENTITY_NAME Lcom/mojang/datafixers/DSL$TypeReference;
+		COMMENT A type reference which refers to a entity's identifier.
```
Comment (YanisBft) [inline mappings/net/minecraft/datafixer/TypeReferences.mapping]:
```suggestion
		COMMENT A type reference which refers to an entity's identifier.
```

File: mappings/net/minecraft/entity/Entity.mapping [hunk 15]
```diff
@@ -529,7 +534,9 @@ CLASS net/minecraft/class_1297 net/minecraft/entity/Entity
 	METHOD method_5793 removePassenger (Lnet/minecraft/class_1297;)V
 		ARG 1 passenger
 	METHOD method_5794 isConnectedThroughVehicle (Lnet/minecraft/class_1297;)Z
+		COMMENT Checks this entity and another entity share the same root vehicle.
```
Comment (YanisBft) [inline mappings/net/minecraft/entity/Entity.mapping]:
```suggestion
		COMMENT Checks if this entity and another entity share the same root vehicle.
```

File: mappings/net/minecraft/nbt/NbtHelper.mapping [hunk 16]
```diff
@@ -23,20 +23,31 @@ CLASS net/minecraft/class_2512 net/minecraft/nbt/NbtHelper
...
 		ARG 3 oldVersion
+			COMMENT the data version of the compound tag
 	METHOD method_10691 toBlockPos (Lnet/minecraft/class_2487;)Lnet/minecraft/class_2338;
 		ARG 0 tag
 	METHOD method_10692 fromBlockPos (Lnet/minecraft/class_2338;)Lnet/minecraft/class_2487;
 		ARG 0 pos
 	METHOD method_10693 update (Lcom/mojang/datafixers/DataFixer;Lnet/minecraft/class_4284;Lnet/minecraft/class_2487;II)Lnet/minecraft/class_2487;
+		COMMENT Uses the data fixer to update a tag.
 		ARG 0 fixer
+			COMMENT the data fixer.
```
Comment (YanisBft) [inline mappings/net/minecraft/nbt/NbtHelper.mapping]:
```suggestion
			COMMENT the data fixer
```

File: mappings/net/minecraft/nbt/NbtHelper.mapping [hunk 17]
```diff
@@ -23,20 +23,31 @@ CLASS net/minecraft/class_2512 net/minecraft/nbt/NbtHelper
...
 		ARG 1 fixTypes
+			COMMENT the fix types
 		ARG 2 tag
+			COMMENT the tag to fix
 		ARG 3 oldVersion
-		ARG 4 currentVersion
+			COMMENT the data version of the compound tag
+		ARG 4 targetVersion
+			COMMENT the data version to update the tag to.
```
Comment (YanisBft) [inline mappings/net/minecraft/nbt/NbtHelper.mapping]:
```suggestion
			COMMENT the data version to update the tag to
```

**Review by YanisBft — APPROVED** [review]

**Pull request merged**